src/index.tsx

```diff
@@ -146,7 +146,7 @@ function hitTest(element: ExcalidrawElement, x: number, y: number): boolean {
     });
 
     return Math.hypot(a * tx - px, b * ty - py) < lineThreshold;
-  } else if (element.type === "rectangle") {
+  } else if (element.type === "rectangle" || element.type === "diamond") {
```

Comment:
The hit test needs to change to follow the lines, otherwise it's going to react to clicking on the rectangle edges but we display the diamond one

Reply:
@vjeux I'm struggling to get the corners of the shape, any help would be appreciated

Reply:
The way the rectangle hit test work is that we check if the distance between each segment is smaller than X.

So what you want is to check the four segments of the diamond:

topEdge to rightEdge
rightEdge to bottomEdge
bottomEdge to leftEdge
leftEdge to topEdge